src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSelector.java

```diff
@@ -0,0 +1,269 @@
+    private int minBitrateBps;
+    private VideoMediaStreamImpl stream;
+    private SinglePacketTransformer packetTransformer;
+    private final Object critSect = new Object();
+    private RemoteBitrateObserver observer;
+    /**
+     * {@code enableUseofAbsSendTime} enables selection of
+     * RemoteBitrateEstimatorAbsSendTime if set to "true"
+     * Default is false
+     */
+    private boolean enableUseOfAbsSendTime;
+
+    public RemoteBitrateEstimatorSelector(RemoteBitrateObserver observe, VideoMediaStreamImpl stream)
+    {
+        this.observer = observe;
+        this.stream = stream;
+        /**
+         * RemoteBitrateEstimatorSingleStream is the Default BitrateEstimator
+         */
+        this.packetTransformer
+                = new RemoteBitrateEstimatorSingleStream(observe);
+        this.minBitrateBps = 0;
+        this.packetsSinceAbsoluteSendTime = 0;
+        setAbsSendTimeUsageConfiguration();
+    }
+    private void PickEstimatorFromHeader(RawPacket packet) {
+        synchronized (critSect) {
+            if (absSendTimeEngine.getAbsoluteSendTimeExtension(packet) != null)
+            {
+                // If we see AST in header, switch RBE strategy immediately.
+                if (!usingAbsoluteSendTime) {
+                    logger.warn("WrappingBitrateEstimator: Switching to" +
+                            " absolute send time RBE.");
+                    usingAbsoluteSendTime = true;
+                    PickEstimator();
+                }
+                packetsSinceAbsoluteSendTime = 0;
+            } else {
+                // When we don't see AST, wait for a few packets before going
+                // back to TOF.
+                if (usingAbsoluteSendTime) {
+                    ++packetsSinceAbsoluteSendTime;
+                    if (packetsSinceAbsoluteSendTime
+                            >= kTimeOffsetSwitchThreshold) {
+                        logger.info("WrappingBitrateEstimator: Switching to " +
+                                "transmission time offset RBE.");
+                        usingAbsoluteSendTime = false;
+                        PickEstimator();
+                    }
+                }
+            }
+        }
+    }
+
+    public void setAbsSendTimeUsageConfiguration()
+    {
+        ConfigurationService cfg = LibJitsi.getConfigurationService();
+
+        if (cfg == null)
+        {
+            logger.warn("FORGETTING about checking AbsSendTime because "
+                    + "the configuration service was not found.");
+            return;
+        }
+        //@Todo set default value to false. Allow true for test.
+        enableUseOfAbsSendTime
+                = cfg.getBoolean(ENABLE_USE_OF_ABS_SEND_TIME, false);
+    }
+
+    /**
+     * Selects either to use RBE_abs_send_time or RBE_single_stream
+     */
+    private void PickEstimator()
+    {
+        synchronized (critSect)
+        {
+            if (usingAbsoluteSendTime)
```

Comment:
Does this mean the synchronized construct can be omitted from the `else ` block?